examples/kubernetes/kubernetes-config/main.tf

```diff
@@ -0,0 +1,147 @@
+terraform {
+  required_providers {
+    digitalocean = {
+      source  = "digitalocean/digitalocean"
+      version = ">= 2.4.0"
+    }
+    kubernetes = {
+      source = "hashicorp/kubernetes"
+      version = ">= 2.0.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">= 2.0.1"
+    }
+  }
+}
+
+data "digitalocean_kubernetes_cluster" "primary" {
+  name = var.cluster_name
+}
+
+resource "local_file" "kubeconfig" {
+  depends_on = [var.cluster_id]
```

Comment:
This establishes a dependency on the doks-cluster module, because referencing the `data.digitalocean_kubernetes_cluster` (instead of the `resource`) decoupled them?

Reply:
Right. In this example, the aim is to still do something that can be created in a single apply. So we do want to establish a dependency here, but we do want to decouple the actual credentials by using the data-source.